default.nix

```diff
@@ -15,42 +15,66 @@ rec {
     in
       pkg.offline_cache;
 
-  buildYarnPackageDeps = { name, packageJson, yarnLock, yarnNix ? null }:
+  buildYarnPackageDeps = { name, packageJson, yarnLock, yarnNix ? null, pkgConfig ? {} }:
     let
       yarnNix_ =
         if yarnNix == null then (generateYarnNix yarnLock) else yarnNix;
       offlineCache =
         loadOfflineCache yarnNix_;
+      moreBuildInputs = (pkgs.lib.flatten (builtins.map (key:
+        pkgConfig.${key} . buildInputs or []
+      ) (builtins.attrNames pkgConfig)));
+      postInstall = (builtins.map (key:
+        if (pkgConfig.${key} ? "postInstall") then
+          ''
+            test -e node_modules/${key} && (
+              cd node_modules/${key}
+              ${pkgConfig.${key}.postInstall}
+            )
+          ''
+        else
+          ""
+      ) (builtins.attrNames pkgConfig));
     in
     stdenv.mkDerivation {
       name = "${name}-modules";
 
       phases = ["buildPhase"];
-      buildInputs = [ yarn nodejs-6_x ];
+      buildInputs = [ yarn nodejs ] ++ moreBuildInputs;
 
       buildPhase = ''
+        set -x
         # Yarn writes cache directories etc to $HOME.
         export HOME=`pwd`/yarn_home
 
         cp ${packageJson} ./package.json
         cp ${yarnLock} ./yarn.lock
 
-        yarn config set yarn-offline-mirror ${offlineCache}
+        yarn config --offline set yarn-offline-mirror ${offlineCache}
 
         # Do not look up in the registry, but in the offline cache.
         # TODO: Ask upstream to fix this mess.
-        sed -i 's/https:\/\/registry.yarnpkg.com\/.*\/-\///' yarn.lock
-        yarn --offline --frozen-lockfile
+        sed -i -E 's|^(\s*resolved\s*")https?://.*/|\1|' yarn.lock
```

Comment:
We're using a private npm registry, the lines look like this:
`  resolved "http://10.11.22.33:8080/a/accepts/_attachments/accepts-1.3.3.tgz#c3ca7434938648c3e0d9c1e328dd68b622c284ca"`, this regex handles both private and public.